metrics/src/main/java/io/opencensus/metrics/LabelValue.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.ExperimentalApi;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * The value of a {@code Label} associated with a {@code TimeSeries}.
+ *
+ * @since 0.15
+ */
+@ExperimentalApi
+@Immutable
+@AutoValue
+public abstract class LabelValue {
+
+  LabelValue() {}
+
+  /**
+   * Creates a {@link LabelValue}.
+   *
+   * @param value the value of a {@code Label}.
+   * @return a {@code LabelValue}.
+   * @since 0.15
+   */
+  public static LabelValue create(String value) {
+    return new AutoValue_LabelValue(value, true);
+  }
+
+  /**
+   * Creates a {@link LabelValue} with unset value.
+   *
+   * @return a {@code LabelValue} with unset value.
+   * @since 0.15
+   */
+  public static LabelValue create() {
```

Comment:
Could we just make the String nullable in the previous method?  I think that would be simpler, and easier to use if the value is already stored as a nullable String.

Reply:
Then we won't be able to distinguish the case with set empty value and unset empty value:
`{value = "", hasValue = true}` vs `{value = "", hasValue = false}`

Do you think it's better to disallow `{value = "", hasValue = false}` and have only `{value = null, hasValue = false}` instead?

Reply:
I thought that we ignore the `value` field when `has_value` is false in the LabelValue proto, because it is simulating an optional string value.  I think we can represent the same thing in Java with a single `@Nullable String` field.

Reply:
Make sense, done.

Reply:
hasValue() can be removed now, too.